src/engine/strat_engine/backstore/crypt.rs

```diff
@@ -674,72 +782,22 @@ fn check_luks2_token(device: &mut CryptDevice) -> Result<()> {
 /// Validate that the LUKS2 token is present and valid
 ///
 /// May not be necessary. See the comment above the invocation.
-fn luks2_token_type_is_valid(json: &serde_json::Value) -> bool {
+fn luks2_token_type_is_valid(json: &Value) -> bool {
     json.get(TOKEN_TYPE_KEY)
         .and_then(|type_val| type_val.as_str())
         .map(|type_str| type_str == LUKS2_TOKEN_TYPE)
         .unwrap_or(false)
 }
 
 /// Validate that the Stratis token is present and valid
-fn stratis_token_is_valid(json: &serde_json::Value) -> bool {
+fn stratis_token_is_valid(json: &Value) -> bool {
     debug!("Stratis LUKS2 token: {}", json);
 
-    let type_valid = json
-        .get(TOKEN_TYPE_KEY)
-        .and_then(|type_val| type_val.as_str())
-        .map(|type_str| type_str == STRATIS_TOKEN_TYPE)
-        .unwrap_or(false);
-    if !type_valid {
-        debug!("Type value is invalid");
-    }
-
-    let keyslots_valid = json
-        .get(TOKEN_KEYSLOTS_KEY)
-        .and_then(|arr| arr.as_array())
-        .map(|arr| arr.is_empty())
-        .unwrap_or(false);
-    if !keyslots_valid {
-        debug!("Keyslots value is invalid");
+    let result = StratisLuks2Token::try_from(json);
+    if let Err(ref e) = result {
+        debug!("Conversion of Stratis JSON token failed with error: {}", e);
```

Comment:
Please reword slightly:

```"Interpretation of LUKS2 token in JSON format as Stratis token failed with error: {}", e```

Otherwise it looks like it's reporting an unexpected error, rather than something that could likely happen under normal operation.

Reply:
I've reworded slightly from what you had here but I think I've addressed the concern you had.